bin/auth_ejabberd.php

```diff
@@ -55,11 +55,15 @@
 require dirname(__DIR__) . '/vendor/autoload.php';
 
 $basedir = BasePath::create(dirname(__DIR__), $_SERVER);
-$configLoader = new Config\ConfigCacheLoader($basedir);
-$config = Factory\ConfigFactory::createCache($configLoader);
-$logger = Factory\LoggerFactory::create('auth_ejabberd', $config);
+$configLoader = new Cache\ConfigCacheLoader($basedir);
+$configCache = Factory\ConfigFactory::createCache($configLoader);
+Factory\DBFactory::init($configCache, $_SERVER);
+$config = Factory\ConfigFactory::createConfig($configCache);
+// needed to call PConfig::init()
+Factory\ConfigFactory::createPConfig($configCache);
+Factory\LoggerFactory::create('auth_ejabberd', $config);
```

Comment:
Please move this whole sequence into a static factory method. This is getting way more than you can safely duplicate in each endpoint.

Reply:
There's already a feature branch where I use the DI-library exactly for this sequence, so yes of course, I'll move it temporary in a static method :)